packages/clerk-js/src/ui/elements/Avatar.tsx

```diff
@@ -65,23 +65,50 @@ export const Avatar = (props: AvatarProps) => {
   // TODO: Revise size handling. Do we need to be this dynamic or should we use the theme instead?
   return (
     <Flex
-      elementDescriptor={[boxElementDescriptor || descriptors.avatarBox]}
+      elementDescriptor={[boxElementDescriptor, descriptors.avatarBox]}
       sx={[
-        theme => ({
+        t => ({
           flexShrink: 0,
-          borderRadius: rounded ? theme.radii.$circle : theme.radii.$md,
+          borderRadius: rounded ? t.radii.$circle : t.radii.$md,
           overflow: 'hidden',
-          width: size(theme),
-          height: size(theme),
-          border: theme.borders.$normal,
-          borderColor: theme.colors.$avatarBorder,
-          backgroundColor: theme.colors.$avatarBackground,
+          width: 'var(--cl-avatar-size)',
+          height: 'var(--cl-avatar-size)',
+          backgroundColor: t.colors.$avatarBackground,
           backgroundClip: 'padding-box',
+          position: 'relative',
+          boxShadow: 'var(--cl-shimmer-hover-shadow)',
+          transition: `box-shadow ${t.transitionDuration.$slower} ${t.transitionTiming.$easeOut}`,
+          '--cl-avatar-size': size(t),
         }),
         sx,
       ]}
     >
       {ImgOrFallback}
+
+      <Box
```

Comment:
Lets add a short comment explaining that this Box renders the shimmer + the border animation (is that the :after element?)

Reply:
👍